<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1030</width>
    <height>900</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow, QWidget#centralwidget, QStackedWidget{
	background-color: rgb(54, 60, 75);
}

QLabel {
color: rgb(255, 255, 255);
}

QPushButton {
	background-color: rgb(255,255,255);
	color: rgb(54,60,75);
}

#ip_page, #arp_page, #dns_page {
	background-color: rgb(54, 60, 75);
}

#scrollAreaWidgetContents {
	background-color: rgb(82, 88, 104);
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2" stretch="5,1">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,5,0,5,0">
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QVBoxLayout" name="left_side" stretch="1,0,5">
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <layout class="QVBoxLayout" name="protocol_selection_container" stretch="1,1">
          <item>
           <widget class="QLabel" name="protocol_selection_title">
            <property name="font">
             <font>
              <pointsize>24</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Protocol Selection</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,1">
            <item>
             <widget class="QLabel" name="protocol_label">
              <property name="text">
               <string>Protocol Type</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="protocol_input">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <item>
               <property name="text">
                <string>IP</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ARP</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>DNS</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="first_spacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QVBoxLayout" name="protocol_details_container">
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QStackedWidget" name="protocol_types_configuration_pages">
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="IPConfigurationWidget" name="ip_page">
             <property name="font">
              <font>
               <family>Calibri</family>
               <pointsize>20</pointsize>
              </font>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout" stretch="1,1,1,1,1">
              <property name="spacing">
               <number>12</number>
              </property>
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QLabel" name="protocol_selection_title_3">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="font">
                 <font>
                  <pointsize>20</pointsize>
                 </font>
                </property>
                <property name="text">
                 <string>IP Packet Configuration</string>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QHBoxLayout" name="source_ip_address_layout">
                <item>
                 <widget class="QLabel" name="source_ip_address_label">
                  <property name="font">
                   <font>
                    <pointsize>14</pointsize>
                   </font>
                  </property>
                  <property name="text">
                   <string>Source IP Address</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="IPAddressInput" name="source_ip_address_input">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="ip_payload_layout">
                <item>
                 <widget class="QLabel" name="ip_payload_label">
                  <property name="font">
                   <font>
                    <pointsize>14</pointsize>
                   </font>
                  </property>
                  <property name="text">
                   <string>Payload</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="ip_payload_input">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="destination_ip_address_layout">
                <item>
                 <widget class="QLabel" name="destination_ip_address_label">
                  <property name="font">
                   <font>
                    <pointsize>14</pointsize>
                   </font>
                  </property>
                  <property name="text">
                   <string>Destination IP Address</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="IPAddressInput" name="destination_ip_address_input">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="number_of_ip_packets_layout">
                <item>
                 <widget class="QLabel" name="number_of_ip_packets_label">
                  <property name="font">
                   <font>
                    <pointsize>14</pointsize>
                   </font>
                  </property>
                  <property name="text">
                   <string>How many of this packet?</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="number_of_ip_packets">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="page_2"/>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QVBoxLayout" name="right_side" stretch="1,11">
        <property name="spacing">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="packets_to_send_title">
          <property name="font">
           <font>
            <pointsize>24</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Packets To Send</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QScrollArea" name="packets_to_send_scroll_container">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
          </property>
          <widget class="QWidget" name="scroll_area_contents">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>429</width>
             <height>625</height>
            </rect>
           </property>
           <property name="styleSheet">
            <string notr="true">background-color: rgba(82, 88, 104, 1)</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_11">
            <property name="spacing">
             <number>0</number>
            </property>
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <layout class="QVBoxLayout" name="to_send_packets_list">
              <property name="spacing">
               <number>10</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,1,0,1,0">
      <item>
       <spacer name="horizontalSpacer_5">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="add_packet_button">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>100</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>20</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Add Packets</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_6">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="send_packets_button">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>100</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>20</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Send packets</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1030</width>
     <height>31</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QMenuBar {
	background-color: rgb(100, 100, 100);
	font: 75 16pt &quot;MS Shell Dlg 2&quot;;
	color: rgb(255, 255, 255);
}
</string>
   </property>
   <widget class="QMenu" name="menuPacket_Generation">
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="title">
     <string>Packet Generation</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuSend_Summary">
    <property name="title">
     <string>Send Summary</string>
    </property>
   </widget>
   <addaction name="menuPacket_Generation"/>
   <addaction name="menuSend_Summary"/>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>IPAddressInput</class>
   <extends>QLineEdit</extends>
   <header>ip_4_address_input.h</header>
  </customwidget>
  <customwidget>
   <class>IPConfigurationWidget</class>
   <extends>QWidget</extends>
   <header location="global">ip_packet_configuration.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
